package.json

```diff
@@ -52,30 +52,34 @@
     "url": "https://github.com/alex3165/react-mapbox-gl/issues"
   },
   "homepage": "https://github.com/alex3165/react-mapbox-gl#readme",
-  "dependencies": {
+  "dependencies": {},
+  "peerDependencies": {
     "deep-equal": "^1.0.1",
     "mapbox-gl": "^0.35.0",
+    "react": "^15.4.2",
+    "react-dom": "^15.4.2",
     "reduce-object": "^0.1.3",
     "supercluster": "^2.3.0"
   },
-  "peerDependencies": {
-    "react": "^15.4.2",
-    "react-dom": "^15.4.2"
-  },
   "devDependencies": {
     "@types/core-js": "^0.9.41",
+    "@types/deep-equal": "^1.0.0",
     "@types/enzyme": "^2.7.7",
     "@types/geojson": "^1.0.1",
     "@types/jest": "^19.2.2",
+    "@types/mapbox-gl": "^0.30.0",
```

Comment:
I compared [your defs](https://raw.githubusercontent.com/alex3165/react-mapbox-gl/master/src/util/mapbox-type.d.ts) and [definitelyTyped](https://raw.githubusercontent.com/DefinitelyTyped/DefinitelyTyped/master/types/mapbox-gl/index.d.ts):  
they are almost identical, except for just a couple of lines... 

I think you could have a much shorter `mapbox-type.d.ts` by extending types where needed or the namespace...

```ts
declare namespace mapboxgl {
  export class MEvented extends Evented {
    on(type: string, listener: Function): this;
    on(type: string, layerId: string, listener: Function): this;
  }

  export class MMap extends Map {
    setFilter(layer: string, filter?: any[]): this;
  }
}
```